src/morx.rs

```diff
@@ -0,0 +1,1824 @@
+                    }
+                    (NUMBER_SPACING_TYPE, PROPORTIONAL_NUMBERS) => {
+                        feature_mask.contains(FeatureMask::PNUM)
+                    }
+                    (NUMBER_SPACING_TYPE, TABULAR_NUMBERS) => {
+                        feature_mask.contains(FeatureMask::TNUM)
+                    }
+                    (FRACTION_TYPE, FRACTIONS_DIAGONAL) => feature_mask.contains(FeatureMask::FRAC),
+                    (FRACTION_TYPE, FRACTIONS_STACKED) => feature_mask.contains(FeatureMask::AFRC),
+                    (FRACTION_TYPE, NO_FRACTIONS) => {
+                        !feature_mask.contains(FeatureMask::FRAC)
+                            && !feature_mask.contains(FeatureMask::AFRC)
+                    }
+                    (VERTICAL_POSITION_TYPE, ORDINALS) => feature_mask.contains(FeatureMask::ORDN),
+                    (TYPOGRAPHIC_EXTRAS_TYPE, SLASHED_ZERO_ON) => {
+                        feature_mask.contains(FeatureMask::ZERO)
+                    }
+                    (TYPOGRAPHIC_EXTRAS_TYPE, SLASHED_ZERO_OFF) => {
+                        !feature_mask.contains(FeatureMask::ZERO)
+                    }
+                    (LOWERCASE_TYPE, LOWERCASE_SMALL_CAPS) => {
+                        feature_mask.contains(FeatureMask::SMCP)
+                            || feature_mask.contains(FeatureMask::C2SC)
+                    }
+                    (UPPERCASE_TYPE, UPPERCASE_SMALL_CAPS) => {
+                        feature_mask.contains(FeatureMask::C2SC)
+                    }
+                    (LIGATURE_TYPE, COMMON_LIGATURES_ON) => {
+                        feature_mask.contains(FeatureMask::LIGA)
+                    }
+                    (LIGATURE_TYPE, COMMON_LIGATURES_OFF) => {
+                        !feature_mask.contains(FeatureMask::LIGA)
+                    }
+                    (LIGATURE_TYPE, HISTORICAL_LIGATURES_ON) => {
+                        feature_mask.contains(FeatureMask::HLIG)
+                    }
+                    (LIGATURE_TYPE, HISTORICAL_LIGATURES_OFF) => {
+                        !feature_mask.contains(FeatureMask::HLIG)
+                    }
+                    (LIGATURE_TYPE, CONTEXTUAL_LIGATURES_ON) => {
+                        feature_mask.contains(FeatureMask::CLIG)
+                    }
+                    (LIGATURE_TYPE, CONTEXTUAL_LIGATURES_OFF) => {
+                        !feature_mask.contains(FeatureMask::CLIG)
+                    }
+                    _ => false,
+                };
+
+                if apply {
+                    subfeature_flags =
+                        (subfeature_flags & entry.disable_flags) | entry.enable_flags;
+                }
+            }
+        }
+    }
+    Ok(subfeature_flags)
+}
+
+//------------------------------------ Ligature Substitution Test ----------------------------------------------------------
+pub fn morx_ligature_test<'a>(scope: ReadScope<'a>) -> Result<(), ParseError> {
```

Comment:
Tests should have `test` attribute so they are run by `cargo test`,

```suggestion
#[test]
pub fn morx_ligature_test<'a>(scope: ReadScope<'a>) -> Result<(), ParseError> {
```